src/uagents/context.py

```diff
@@ -97,6 +99,22 @@ def get_message_protocol(self, message_schema_digest) -> Optional[str]:
                     return protocol_digest
         return None
 
+    def get_agents_by_protocol(self, protocol_digest: str) -> list:
+        if not isinstance(protocol_digest, str) or not protocol_digest.startswith(
+            "proto:"
+        ):
+            self.logger.error(f"Invalid protocol digest: {protocol_digest}")
+            return []
```

Comment:
This doesn't really feel like a it should be a silent error to me. If the user is not providing the right values this seems like this should `raise` some sort of `ValueError` - that would be clearer I think